src/main/java/com/spikes2212/dashboard/AutoChooser.java

```diff
@@ -0,0 +1,189 @@
+     *               the name for the previous command
+     * @throws IllegalArgumentException when an even index is a {@link String}, an odd index is a {@link Command} or
+     * when the options length isn't even.
+     */
+    public AutoChooser(Command defaultOption, String defaultOptionName, Object... options) {
+        this(new RootNamespace(DEFAULT_NAMESPACE_NAME), defaultOption, defaultOptionName, INCREMENTATION_VALUE_ONE, options);
+    }
+
+    private AutoChooser(Namespace namespace, Command defaultOption, String defaultOptionName,
+                        int incrementationValue, Object... options) {
+        if (options.length % 2 != 0) throw new IllegalArgumentException("The options' length isn't even.");
+        this.namespace = namespace;
+        setDefaultOption(defaultOptionName, defaultOption);
+        Command command;
+        String name;
+        int num;
+        for (int i = 0; i < options.length; i += 2) {
+            if (options[i] instanceof Command) {
+                command = (Command) options[i];
+            } else {
+                num = i + incrementationValue;
+                throw new IllegalArgumentException("The " + num + getCorrectSuffix(num) + " argument is not a Command.");
+            }
+            if (options[i + 1] instanceof String) {
+                name = (String) options[i + 1];
+            } else {
+                num = i + incrementationValue + 1;
+                throw new IllegalArgumentException("The " + num + getCorrectSuffix(num) + " argument is not a String.");
+            }
+            addOption(name, command);
+        }
+        putOnShuffleboard();
+    }
+
+    /**
+     * Adds the given command with the given name as an option to this {@link AutoChooser}. <br>
+     * In case this name already exists, a number will be added to differentiate between the options.
+     */
+    @Override
+    public void addOption(String name, Command command) {
+        if (!addName(name)) {
+            String originalName = name;
+            int i = 2;
+            name += " " + i;
+            while (!addName(name)) {
+                i++;
+                name = originalName + " " + i;
+            }
+        }
+        super.addOption(name, command);
+    }
+
+    /**
+     * Schedules the selected command and closes this {@link AutoChooser}.
+     */
+    public void schedule() {
+        getSelected().schedule();
+        close();
+    }
+
+    /**
+     * @param num a positive number
+     * @return the correct suffix for this number
+     */
+    private String getCorrectSuffix(int num) {
+        List<String> suffixes = List.of("th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th");
+        switch (num % 100) {
+            case 11:
+            case 12:
+            case 13:
+                return "th";
+            default:
+                return suffixes.get(num % 10);
+        }
+    }
+
+    private boolean addName(String name) {
```

Comment:
I don't really get why that's necessary but fine

Reply:
it's just nicer to work with